packages/core/src/operator/listenForChallenges.ts

```diff
@@ -11,40 +9,45 @@ import { resilientEventListener } from "../utils/resilientEventListener.js";
  * @param callback - The callback function to be triggered when ChallengeSubmitted event is emitted.
  * @returns A function that can be called to stop listening for the event.
  */
-export function listenForChallenges(callback: (challengeNumber: bigint, challenge: Challenge, event: any) => void): () => void {
-
-    // create a map to keep track of challengeNumbers that have called the callback
+export function listenForChallenges(callback: (challengeNumber: bigint, challenge: Challenge, event: any) => void, onError?: (error: Error) => void): () => void {
     const challengeNumberMap: { [challengeNumber: string]: boolean } = {};
 
-    // listen for the ChallengeSubmitted event
     const listener = resilientEventListener({
         rpcUrl: config.arbitrumOneWebSocketUrl,
         contractAddress: config.refereeAddress,
         abi: RefereeAbi,
         eventName: "ChallengeSubmitted",
         log: console.info,
         callback: async (log, error) => {
-
-            if(error){
-                //TODO hanlde on error
+            if (error) {
+                console.error(error);
+                if(!onError) return;
+                // Call the onError function with the caught error
+                onError(error instanceof Error ? error : new Error(String(error)));
                 return;
             }
-
-            const challengeNumber = BigInt(log?.args[0]);
-
-            // if the challengeNumber has not been seen before, call the callback and add it to the map
-            if (!challengeNumberMap[challengeNumber.toString()]) {
-                challengeNumberMap[challengeNumber.toString()] = true;
-
-                // lookup the challenge
-                const challenge = await getChallenge(challengeNumber);
-
-                void callback(challengeNumber, challenge, log);
+            try {
+                // Uncomment the following line to test the onError function
+                const challengeNumber = BigInt(log?.args[0]);
+
+                // if the challengeNumber has not been seen before, call the callback and add it to the map
+                if (!challengeNumberMap[challengeNumber.toString()]) {
+                    challengeNumberMap[challengeNumber.toString()] = true;
+
+                    // lookup the challenge
+                    const challenge = await getChallenge(challengeNumber);
+
+                    void callback(challengeNumber, challenge, log);
+                }
+            } catch (err) {
+                console.error(err);
```

Comment:
Please extend this error log with additional context as well as handle the error object. This could get printed in the operator and would be a log the users have never seen since.

Reply:
Removed!